### DISCUSSION TITLE
[Q&A] - Global and Local model for xgboost histogram-based

### DISCUSSION THREAD

**Discussion by ramiro350** [NVIDIA/NVFlare#2703] [Q&A] [answered]
### Python version (`python3 -V`)

3.10.12

### NVFlare version (`python3 -m pip list | grep "nvflare"`)

2.4.0

### NVFlare branch (if running examples, please use the branch that corresponds to the NVFlare version, `git branch`)

main

### Operating system

Ubuntu 22.04

### Have you successfully run any of the following examples?

- [ ] hello-numpy-sag with simulator
- [ ] hello-pt with simulator
- [ ] hello-numpy-sag with POC
- [ ] hello-pt with POC

### Please describe your question

Hello guys, i've been running the xgboost histogram-based example(this one https://github.com/NVIDIA/NVFlare/tree/main/examples/advanced/xgboost/histogram-based) i'm running it in real world, one server and one client, i've launched on the admin cmd and everything went well i believe, i was checking in the client and in the job directory i've found the test.model which i believe to be the local model, my question is, is this actually the local model ? and where is the global model stored in the server ?

**Comment by YuanTingHsieh** [comment]
@ramiro350 thanks for the interest!

For your question:

is this actually the local model ?
and where is the global model stored in the server ?


@ZiyueXu77 can you help answer these questions?
The local model was saved in here: https://github.com/NVIDIA/NVFlare/blob/2.4/nvflare/app_opt/xgboost/histogram_based/executor.py#L288
Is the global model the same as the local model we get in federated xgboost (the insecure version) after training is done?

**Comment by ZiyueXu77** [comment, marked as answer]
In this case (histogram-based) server only performs necessary communication during tree-building, so it does not hold any model. The outcome model is the same across all clients and will be stored locally.
